### ISSUE TITLE
Hanging filehandle

### ISSUE THREAD

**Issue by alhirzel** [fhs/ZipFile.jl#98]
https://github.com/fhs/ZipFile.jl/blob/7dbf67d22a4f52cdc6d8225a6bfa9dbd844ff990/src/ZipFile.jl#L121

I noticed that this creates a hanging file handle, and in some cases the file handle can get closed prior to actually consuming data from the `ZipFile.Reader`.

**Comment by JoaoAparicio** [comment]
Would you like to have a look at https://github.com/fhs/ZipFile.jl/issues/14 and check if it's related?

**Comment by alhirzel** [comment]
Yes. The discussion in that issue are very related! It may be worth closing this issue and addressing that one instead, as it has more context.
